src/js/media-theme-element.js

```diff
@@ -0,0 +1,99 @@
+import { Window as window } from './utils/server-safe-globals.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import { TemplateInstance } from './utils/template-parts.js';
+import { processor } from './utils/template-processor.js';
+
+// Export Template parts for players.
+export * from './utils/template-parts.js';
+
+export class MediaThemeElement extends window.HTMLElement {
+  static observedAttributes = ['template'];
+  static processor = processor;
+
+  renderRoot;
+  renderer;
+  #template;
+
+  constructor() {
+    super();
+    this.renderRoot = this.attachShadow({ mode: 'open' });
+
+    const observer = new MutationObserver(() => this.render());
+    observer.observe(this, { attributes: true });
+  }
+
+  get mediaController() {
+    // Expose the media controller if API access is needed
+    return this.renderRoot.querySelector('media-controller');
+  }
+
+  get template() {
+    const templateId = this.getAttribute('template');
+    if (templateId) {
+      const template = this.getRootNode()?.getElementById(templateId);
+      if (template) return template;
+    }
+    return this.#template;
+  }
+
+  set template(element) {
+    this.#template = element;
```

Comment:
idea(non-blocking): We should probably figure out some way of not requiring an actual DOM element instance of the template having to exist here, otherwise the definition _**has to**_ be explicitly declared on the DOM in the same context/page as the use of it (at least without also including some build step/additional work). This means we can't easily **_define_** a theme template in one place and _**use it**_ in another place. A simple solution for this would be to support a string or element for `set template` and do something like https://stackoverflow.com/questions/494143/creating-a-new-dom-element-from-an-html-string-using-built-in-dom-methods-or-pro/35385518#35385518 (I know, a template used to create a template)

Reply:
hmm, I'm not fully convinced this is needed. this was mainly added as a secondary way to set a template element.
first being the `template` attribute that is a `id` to query the DOM.

they can always create a template element if they have the html string right.
maybe as a convenience method setting / getting `templateHTML`